napari/plugins/_tests/test_utils.py

```diff
@@ -95,14 +106,16 @@ def test_get_filename_patterns_fake_plugin():
     assert len(get_filename_patterns_for_reader('gibberish')) == 0
 
 
-def test_get_filename_patterns(mock_npe2_pm, tmp_reader):
-    fake_reader = tmp_reader(mock_npe2_pm, 'fake-reader', ['*.tif'])
+def test_get_filename_patterns(tmp_plugin: DynamicPlugin):
+    @tmp_plugin.contribute.reader(filename_patterns=['*.tif'])
+    def read_tif(path):
```

Comment:
It would be nice if we could easily add contributions without defining these no-op functions... Not sure how or if that would work

Reply:
yeah, I tried doing something like:
```
tmp_plugin.contribute.reader(filename_patterns=['*.tif'])(lambda p: None)
```

but it (smartly) yells at you for trying to contribute a function with `<locals>` in the name :joy:.  I think it's a small price to pay.  More explicit, if perhaps a bit verbose

Reply:
Interesting I wonder if that's what was going wrong when I was trying to give my `tmp_reader` fixture a simple read function - I wasn't defining a lambda but I was defining it within a nested function... Anyway, that's not a big deal so I think leave it as is is great